### PR TITLE
xtensa/esp32: timer driver refactor

### PR DESCRIPTION
**Pull request by saramonteiro**
## Summary
It applies several improvements that were done while developing the timer driver for the ESP32-C3.
It removes unused code.
It Improves some logic in the isr attach function and in the prescaler calculation.

## Impact

All esp32 timer users.

## Testing

It was tested using the timer example for the 4 timers.

### CODE REVIEW

File: arch/xtensa/src/esp32/esp32_tim.c [hunk 1]
```diff
@@ -73,35 +71,28 @@ static void esp32_tim_modifyreg32(FAR struct esp32_tim_dev_s *dev,
...
+static void esp32_tim_reload_now(FAR struct esp32_tim_dev_s *dev);
+static void esp32_tim_getalarmvalue(FAR struct esp32_tim_dev_s *dev,
+                                    uint64_t *value);
+static void esp32_tim_setalarmvalue(FAR struct esp32_tim_dev_s *dev,
+                                    uint64_t value);
+static void esp32_tim_setalarm(FAR struct esp32_tim_dev_s *dev, bool enable);
+static void esp32_tim_setautoreload(FAR struct esp32_tim_dev_s *dev,
+                                    bool enable);
+static  int esp32_tim_setisr(FAR struct esp32_tim_dev_s *dev, xcpt_t handler,
+                             FAR void * arg);
```
Comment (acassis) [inline arch/xtensa/src/esp32/esp32_tim.c:92]: please remove space "* arg"

File: arch/xtensa/src/esp32/esp32_tim.c [hunk 2]
```diff
@@ -252,13 +248,10 @@ static void esp32_tim_modifyreg32(FAR struct esp32_tim_dev_s *dev,
  *
  ****************************************************************************/
 
-static int esp32_tim_start(FAR struct esp32_tim_dev_s *dev)
+static void esp32_tim_start(FAR struct esp32_tim_dev_s *dev)
```
Comment (acassis) [inline arch/xtensa/src/esp32/esp32_tim.c:251]: Why to change it to void? The timer API is expecting a value:
```
    case TCIOC_START:
      {
        /* Start the timer, resetting the time to the current timeout */

        if (lower->ops->start)
          {
            ret = lower->ops->start(lower);
          }
```

**Review by acassis — CHANGES_REQUESTED** [review]

File: arch/xtensa/src/esp32/esp32_tim.c [hunk 3]
```diff
@@ -269,13 +262,10 @@ static int esp32_tim_start(FAR struct esp32_tim_dev_s *dev)
  *
  ****************************************************************************/
 
-static int esp32_tim_stop(FAR struct esp32_tim_dev_s *dev)
+static void esp32_tim_stop(FAR struct esp32_tim_dev_s *dev)
```
Comment (acassis) [inline arch/xtensa/src/esp32/esp32_tim.c:265]: Ditto, keep is as int

File: arch/xtensa/src/esp32/esp32_tim.c [hunk 4]
```diff
@@ -286,98 +276,12 @@ static int esp32_tim_stop(FAR struct esp32_tim_dev_s *dev)
  *
  ****************************************************************************/
 
-static int esp32_tim_clear(FAR struct esp32_tim_dev_s *dev)
+static void esp32_tim_clear(FAR struct esp32_tim_dev_s *dev)
```
Comment (acassis) [inline arch/xtensa/src/esp32/esp32_tim.c:279]: Ditto, keep is as int

**Comment by acassis** [comment]
I think it is not a good idea to move these functions to void, you could implement some logic testings in the future that you are don't realize now.

File: arch/xtensa/src/esp32/esp32_tim.c (hunk 2 above)
Comment (saramonteiro) [reply arch/xtensa/src/esp32/esp32_tim.c:251]: Please, note that this start is from lower half layer not from `esp32_tim.c`. The start function from lower half is still returning int value.

File: arch/xtensa/src/esp32/esp32_tim.c (hunk 3 above)
Comment (saramonteiro) [reply arch/xtensa/src/esp32/esp32_tim.c:265]: Please, note that this stop is from lower half layer not from `esp32_tim.c`. The stop function from lower half is still returning int value.

Comment (acassis) [reply arch/xtensa/src/esp32/esp32_tim.c:265]: Yes, I noticed it later. But I think it is a good idea to follow the same standard. Also to keep in sync with the other timer drivers.

**Comment by saramonteiro** [comment]
> I think it is not a good idea to move these functions to void, you could implement some logic testings in the future that you are don't realize now.

I don't think it's a good idea to keep unnecessary code.
When it comes the time for developing any kind of testing, we prepare the code for this.

File: arch/xtensa/src/esp32/esp32_tim.c (hunk 2 above)
Comment (acassis) [reply arch/xtensa/src/esp32/esp32_tim.c:251]: > Please, note that this start is from lower half layer not from `esp32_tim.c`. The start function from lower half is still returning int value.

Yes, I noticed it later. But I think it is a good idea to follow the same standard. Also to keep in sync with the other timer drivers.

**Comment by acassis** [comment]
Ok, I will merge, but I think these modifications need to applied to the timer from other arch(es) too

**Review by acassis — APPROVED** [review]

**Pull request merged**